magento/inventory
# Remove erronious check

### Description (*)
The isUseConfigManageStock variable only describes whether or not the stockItemConfiguration uses the default config value or the stockItemConfiguration specific config value for this the isManageStock value. It does not say anything about the value of the manage_stock setting. Since this logic is already correctly implemented in the getManageStock function this line can be completely removed.

### Fixed Issues (if relevant)
As it was the line causes a bug where every PUT call to /products/{sku}/stockItems/{stock_item} endpoint with a use_config_manage_stock value of false will always cause that stockItem to register as in_stock until a reindex is run.

### Manual testing scenarios (*)
Previous behavior:

1. Send a PUT request to the /products/{sku}/stockItems/{stock_item} endpoint with is_in_stock false and use_config_manage_stock false
2. The is_salable value in the inventory_stock_{stock_item} view is 1 where it should be 0
3. Trigger a reindex
4. The is_salable value now reads 1

Correct behavior
1. Send a PUT request to the /products/{sku}/stockItems/{stock_item} endpoint with is_in_stock false and use_config_manage_stock false
2. The is_salable value in the inventory_stock_{stock_item} view is 0

### Questions or comments

As it is right now this call will still set products in_stock when Manage stock is disabled. I don't know enough about that workflow to determine if that is the correct behavior.

### Contribution checklist (*)
 - [x ] Pull request has a meaningful description of its purpose
 - [x ] All commits are accompanied by meaningful commit messages
